Source/WebCore/dom/DocumentFullscreen.cpp

```diff
@@ -133,6 +135,38 @@ void DocumentFullscreen::requestFullscreen(Ref<Element>&& element, FullscreenChe
 {
     auto identifier = LOGIDENTIFIER;
 
+    if (protectedDocument()->quirks().shouldEnterNativeFullscreenWhenCallingElementRequestFullscreenQuirk()) {
```

Comment:
Don't we need to call `completionHandler` here?

Also, I'm not sure whether errors handled in `webkitRequestFullscreen` are supposed to propagate here. For instance, I assume it does its own check for a user gesture, but if that fails then `handleError` won't be called like it is in the equivalent check here.